src/gluonts/torch/distributions/piecewise_linear.py

```diff
@@ -0,0 +1,291 @@
+        )
+
+    @staticmethod
+    def _to_orig_params(
+        slopes: torch.Tensor, knot_spacings: torch.Tensor
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        b = PiecewiseLinear.parametrize_slopes(slopes)
+        knot_positions = PiecewiseLinear.parametrize_knots(knot_spacings)
+        return b, knot_positions
+
+    @staticmethod
+    def parametrize_slopes(slopes: torch.Tensor) -> torch.Tensor:
+        slopes_parametrized = torch.diff(slopes, dim=-1)
+        slopes_parametrized = torch.cat(
+            [slopes[..., 0:1], slopes_parametrized], dim=-1
+        )
+
+        return slopes_parametrized
+
+    @staticmethod
+    def parametrize_knots(knot_spacings: torch.Tensor) -> torch.Tensor:
+        # the last entry is 1 and can be omitted
+        knot_positions = torch.cumsum(knot_spacings, dim=-1)[..., :-1]
+
+        # the first knot pos is 0
+        knot_positions = torch.cat(
+            [torch.zeros_like(knot_positions[..., 0:1]), knot_positions],
+            dim=-1,
+        )
+
+        return knot_positions
+
+    def quantile_internal(
+        self, u: torch.Tensor, dim: Optional[int] = None
+    ) -> torch.Tensor:
+        # output shape = u.shape
+        if dim is not None:
+            # when num_samples!=None
+            #
+            # In testing
+            # dim = 0
+            # u.shape          = (num_samples, *batch_shape)
+            # gamma.shape      = (1, *batch_shape)
+            # other_para.shape = (1, *batch_shape, num_pieces)
+            #
+            # In training, u_tilde is needed to compute CRPS
+            # dim = -2
+            # u.shape          = (*batch_shape, num_pieces)
+            # gamma.shape      = (*batch_shape, 1)
+            # other_para.shape = (*batch_shape, 1, num_pieces)
+
+            gamma = self.gamma.unsqueeze(dim=0 if dim == 0 else -1)
+            knot_positions, b = (
+                self.knot_positions.unsqueeze(dim=dim),
+                self.b.unsqueeze(dim=dim),
+            )
+        else:
+            # when num_sample is None
+            #
+            # In testing
+            # dim = None
+            # u.shape          = (*batch_shape)
+            # gamma.shape      = (*batch_shape)
+            # other_para.shape = (*batch_shape, num_pieces)
+
+            gamma = self.gamma
+            knot_positions, b = self.knot_positions, self.b
+
+        u = u.unsqueeze(-1)
+        u_spline = F.relu(u - knot_positions)
+        quantile = gamma + torch.sum(b * u_spline, dim=-1)
+
+        return quantile
+
+    def quantile(self, u: torch.Tensor) -> torch.Tensor:
+        return self.quantile_internal(u, dim=0)
+
+    def cdf(self, z: torch.Tensor) -> torch.Tensor:
+        r"""
+        compute the quantile levels u_tilde s.t. quantile(u_tilde) = z
+
+        Parameters
+        ----------
+        z
+            observations, shape = gamma.shape = (*batch_size)
+
+        Returns
+        -------
+        torch.Tensor
+            u_tilde
```

Comment:
Better to use a description:

```suggestion
            The quantile levels corresponding to the input observations.
```